src/librbd/cache/rwl/InitRequest.cc

```diff
@@ -0,0 +1,217 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+
+#include "librbd/cache/rwl/InitRequest.h"
+#include "librbd/Utils.h"
+#include "common/dout.h"
+#include "common/errno.h"
+#include "librbd/asio/ContextWQ.h"
+#include "librbd/cache/rwl/ImageCacheState.h"
+#include "librbd/cache/ReplicatedWriteLog.h"
+
+#define dout_subsys ceph_subsys_rbd_rwl
+#undef dout_prefix
+#define dout_prefix *_dout << "librbd::cache::rwl:InitRequest " \
+                           << this << " " << __func__ << ": "
+
+namespace librbd {
+namespace cache {
+namespace rwl {
+
+using util::create_async_context_callback;
+using util::create_context_callback;
+
+template <typename I>
+InitRequest<I>* InitRequest<I>::create(I &image_ctx,
+                                       Context *on_finish) {
+  return new InitRequest(image_ctx, on_finish);
+}
+
+template <typename I>
+InitRequest<I>::InitRequest(I &image_ctx, Context *on_finish)
+  : m_image_ctx(image_ctx),
+    m_on_finish(create_async_context_callback(image_ctx, on_finish)),
+    m_error_result(0) {
+}
+
+template <typename I>
+InitRequest<I>::~InitRequest() {
+}
+
+template <typename I>
+bool InitRequest<I>::is_rwl_enabled() {
+  return m_image_ctx.config.template get_val<bool>("rbd_rwl_enabled");
```

Comment:
Nit: perhaps you can create a `librbd::cache::util::is_rwl_enabled` helper? That would avoid the references throughout the PR to the `rbd_rwl_enabled` config value. In fact, given that one day there should be an SSD persistent cache, maybe it should be an enum config parameter for `rbd_persistent_cache_mode` for `none`, `rwl`, or `ssd` -- and the helper method returns an enum.

Reply:
Moved to src/librbd/cache/Utils.h.